contracts/wallet/WalletFactory.sol

```diff
@@ -200,6 +176,30 @@ contract WalletFactory is Owned, Managed {
         _wallet = address(uint160(uint256(hash)));
     }
 
+    /**
+     * @dev Gets the address of a counterfactual wallet with a first default guardian.
+     * @param _owner The account address.
+     * @param _modules The list of modules.
+     * @param _guardian The guardian address.
+     * @param _salt The salt.
+     * @return the address that the wallet will have when created using CREATE2 and the same input parameters.
+     */
+    function getAddressForCounterfactualWalletWithGuardian(
```

Comment:
If we are going to be deduplicating code, we should do it here as well for `getAddressForCounterfactualWalletWithGuardian` and `getAddressForCounterfactualWallet`

Reply:
Agreed and changed.